meson.build

```diff
@@ -556,6 +556,35 @@ endif
 
 
 
+###############################################################
+# Library: pam
+###############################################################
+
+pamopt = get_option('pam')
+if not pamopt.disabled()
+  pam = dependency('pam', required: pamopt)
+
+  if not pam.found()
+    pam = cc.find_library('pam', required: pamopt, dirs: g_c_lib)
+  endif
+
+  if pam.found()
+    cdata.set('USE_PAM', 1)
+
+    if cc.check_header('security/pam_appl.h', args: g_c_args, dependencies: pam, required: pamopt)
+      cdata.set('HAVE_SECURITY_PAM_APPL_H', 1)
+    elif cc.check_header('pam/pam_appl.h', args: g_c_args, dependencies: pam, required: pamopt)
+      cdata.set('HAVE_PAM_PAM_APPL_H', 1)
+    else
+      error('header file <security/pam_appl.h> or <pam/pam_appl.h> is required for PAM.')
```

Comment:
I think this will error auto when using 'auto' mode, with the library present, but headers not.

Reply:
```meson
if not pamopt.disabled()
  pam = dependency('pam', required: pamopt)

  if not pam.found()
    pam = cc.find_library('pam', required: pamopt, dirs: g_c_lib)
  endif
```

There's a couple awkward points here. You only enter the dependency lookup code if it is not disabled, because you need to do all kinds of fun conditionals. But the first thing you do is look up the pkg-config dependency with `required: pamopt` and possible values of `auto` or `enabled`.

In the latter case, Meson will error out because the pkg-config dependency was not found, and not progress to find_library. Instead, use `required: false`.

If the pkg-config dependency was not found, then look up:
- the library with `required: pamopt`
- the first potential header name with `required: false`
- if the first header is not found -> the second potential header with `required: pamopt`

- If it is still not found, set e.g. a variable `pam_enabled = false`, and rest assured that you do not need an `error()` because if it was enabled, it would have already errored out.

Reply:
FWIW, there's several other dependencies that have "solved" this, so looking at those might be helpful...

Reply:
Yes, I realized that. For example `tcl` will fail because of that reason, I am going create a PR for that tomorrow.